hayha/cloudformation_types.py

```diff
@@ -21,33 +21,151 @@ def __init__(self, outgoing_connection=None, incoming_connection=None, connectio
+            to “A” or if “A” is configured to have an ‘incoming_connection’ from “C” ,
+            that connection will be subject to protection/authentication by resource “B”.
+            In these cases, the minimum security level required to pass protection/authentication
+            by “B” shall be added onto the minimum security level required to reach
+            “C”. Their sum becomes the minimum security level required to reach “A”.
+
+        ‘exit_protection_by’:
+            Description: A key-value property in the parent resource type(“A”) that
+            specifies that all outgoing connections from “A” are protected by the
+            specified resource type (“B”), using “K”, which is a CloudFormation property/key name
+            in “A”. In the context of the Hayha tool/files, “K” refers to its respective
+            CloudFormation value, which is a reference pointer to “B”. If any resource type (“C”)
+            is configured to have an ‘incoming_connection’ from “A” or if “A” is
+            configured to have an ‘outgoing_connection’ to “C”, that connection will
+            be subject to protection/authentication by resource “B”. In these cases, the
+            minimum security level required to pass protection/authentication by “B” shall
+            be added onto the minimum security level required to reach “A”. Their
+            sum becomes the minimum security level required to reach “C”.
+
+        `connection_to_protect`:
+            Description: A key-value property in the parent resource type (“A”) that
+            specifies that “A” protects a specific connection with a specified resource
+            type (“B”) and another resource type (“C”), using “KB” and “KC”, which
+            are CloudFormation properties/key names in “A”. In the context of the
+            Hayha tool/files, “KB” and “KC” refer to their respective values, which
+            are reference pointers to either “B” or  “C”. If configured as ‘managed_protection_flow=[[“B”, “C”]]’,
+            it signifies “B” has an outgoing connection to “C” and that connection
+            will be subject to protection by resource “A”.
+
+        *Still confused about what it means to “have a URL”. I thought it meant having a URL to the AWS documentation about it, but it didn’t make sense because all the resource types are in the AWS documentation. Unsure why we need this property as well*
+        `accessible`:
+            Description: This key-value property indicates the parent resource type
+            is publicly accessible, which means it has a URL.
+
+        *I’m unsure about this definition and am still a little confused why we need this property.*
+        ‘security’:
+            Description: This key-value property indicates that the parent resource
+            type is some sort of security resource type. The parent resource type
+            does not necessarily need to have some protection 
```

Comment:
This property is currently used in the graphs, so the security resources have a red color. It's also used to set the "intrinsic security level" of the resource, meaning that whenever a query goes through it, this enforces some security level. We might want to get rid of it at some point, but we would need to change the implementation a bit.

You could say here that a security resource protects connections between resources according to its configuration and that of the nodes it protects, but there are no connection to or from them.